10.0 Migrate users_ldap_mail module

StephanRozendaal opened the pull request:
Changes include:
 - Migrate users_ldap_mail to 10.0
 - Add README for users_ldap_mail

StephanRozendaal committed:
[ADD] Add README.rst to users_ldap_mail

StephanRozendaal committed:
[MIG] Migrate users_ldap_mail to 10.0

lasley commented on auto_backup/i18n/am.po:
```diff
@@ -0,0 +1,317 @@
+# Translation of Odoo Server.
```
Please delete `auto_backup` files

StephanRozendaal commented:
I removed the "auto_backup" translation commit

StephanRozendaal committed:
[FIX] Fixes in code style in module  users_ldap_mail

Changes in models/users_ldap_mail include:
 - remove unused imports
 - indentation of too long lines

dreispt commented on users_ldap_mail/README.rst:
```diff
@@ -0,0 +1,67 @@
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/{server-tools}/{10.0}
```
Please fix this link.

dreispt commented on users_ldap_mail/__init__.py:
```diff
@@ -19,4 +19,5 @@
 ##############################################################################
 
-from . import users_ldap_model
+from . import models
```
It would be great if you could replace the old large copyright statement with the new leaner one.

dreispt commented on users_ldap_mail/models/users_ldap_model.py:
```diff
@@ -19,50 +19,36 @@
+        help="By default 'cn' is used. "
+             "For ActiveDirectory you might use 'displayName' instead.")
+    mail_attribute = fields.Char(
+        'E-mail attribute', size=64, _defaults='mail',
```
Remove the `size` attribute from `char` fields

dreispt commented on users_ldap_mail/models/users_ldap_model.py:
```diff
@@ -19,50 +19,36 @@
+        ldaps = self.sudo().search([('ldap_server', '!=', False)],
+                                   order='sequence')
+        res = ldaps.read([])
+        return res
```
+1

dreispt reviewed (changes requested):
Looks good, other than the comments made.

StephanRozendaal committed:
[FIX] users_ldap_mail: Fix copyright notice

StephanRozendaal committed:
[FIX] users_ldap_mail: set UTF-8 encoding

users_ldap_mail/models/__init__.py should be UTF-8 encoded

StephanRozendaal committed:
[FIX] users_ldap_mail: remove size argument from Char fields

StephanRozendaal committed:
[FIX] users_ldap_mail: return value can be combined in one line

StephanRozendaal committed:
[FIX] Update read me

Fixes included:
 - Make module name human readable
 - Instead of numbering steps, use #
 - Update broken link

lasley reviewed (approved):
Thanks @StephanRozendaal - 👍 , although some more test coverage would be awesome
